src/tests/unit/init_register/CMakeLists.txt

```diff
@@ -4,14 +4,26 @@ target_link_libraries(initialize_mpi OpenCoarrays)
 add_executable(register register.f90)
 target_link_libraries(register OpenCoarrays)
 
-add_executable(register_rename_me register_rename_me.f90)
-target_link_libraries(register_rename_me OpenCoarrays)
+add_executable(register_vector register_vector.f90)
+target_link_libraries(register_vector OpenCoarrays)
 
-add_executable(register_rename_me_too register_rename_me_too.f90)
-target_link_libraries(register_rename_me_too OpenCoarrays)
+add_executable(register_alloc_vector register_alloc_vector.f90)
+target_link_libraries(register_alloc_vector OpenCoarrays)
+
+add_executable(register_alloc_comp_1 register_alloc_comp_1.f90)
+target_link_libraries(register_alloc_comp_1 OpenCoarrays)
+
+add_executable(register_alloc_comp_2 register_alloc_comp_2.f90)
+target_link_libraries(register_alloc_comp_2 OpenCoarrays)
+
+add_executable(register_alloc_comp_3 register_alloc_comp_3.f90)
+target_link_libraries(register_alloc_comp_3 OpenCoarrays)
 
 add_executable(allocate_as_barrier allocate_as_barrier.f90)
 target_link_libraries(allocate_as_barrier OpenCoarrays)
 
 add_executable(allocate_as_barrier_proc allocate_as_barrier_proc.f90)
 target_link_libraries(allocate_as_barrier_proc OpenCoarrays)
+
+add_executable(check_remote_alloced_comp_1 check_remote_alloced_comp_1.f90)
+target_link_libraries(check_remote_alloced_comp_1 OpenCoarrays)
```

Comment:
@vehre This is failing to compile with GCC 6.2. I'm assuming that this test is only valid for GCC trunk/7.x, correct? If so I'll wrap the CMake for conditional compilation when GCC >= 7.